.circleci/config.yml

```diff
@@ -110,9 +132,54 @@ aliases:
         - /.*-stable/
         - gh-pages
 
+  - &create-ndk-directory
+    |
+      if [[ ! -e /opt/ndk ]]; then
+        sudo mkdir /opt/ndk
+      fi
+      sudo chown ${USER:=$(/usr/bin/id -run)}:$USER /opt/ndk
+
+  # CircleCI does not support interpolating env variables in the environment
+  #  https://circleci.com/docs/2.0/env-vars/#interpolating-environment-variables-to-set-other-environment-variables
+  - &configure-android-path
+    | 
+      echo 'export PATH=${ANDROID_NDK}:~/react-native/gradle-2.9/bin:~/buck/bin:$PATH' >> $BASH_ENV
+      source $BASH_ENV
+
+  - &install-android-packages
+    | 
+      source scripts/circle-ci-android-setup.sh && getAndroidSDK
+
+  - &install-build-dependencies
+    |
+      sudo apt-get update -y
+      sudo apt-get install ant autoconf automake g++ gcc libqt5widgets5 lib32z1 lib32stdc++6  make maven python-dev python3-dev qml-module-qtquick-controls qtdeclarative5-dev file -y
+
+  - &install-android-app-dependencies
+    |
+      buck fetch ReactAndroid/src/test/java/com/facebook/react/modules
+      buck fetch ReactAndroid/src/main/java/com/facebook/react
+      buck fetch ReactAndroid/src/main/java/com/facebook/react/shell
+      buck fetch ReactAndroid/src/test/...
+      buck fetch ReactAndroid/src/androidTest/...
+      ./gradlew :ReactAndroid:downloadBoost :ReactAndroid:downloadDoubleConversion :ReactAndroid:downloadFolly :ReactAndroid:downloadGlog :ReactAndroid:downloadJSCHeaders
+
 defaults: &defaults
   working_directory: ~/react-native
 
+android_defaults: &android_defaults
+  <<: *defaults
+  docker:
+    - image: circleci/android:api-26-alpha
+  resource_class: "large"
+  environment:
+    - TERM: "dumb"
+    - ADB_INSTALL_TIMEOUT: 10
+    - _JAVA_OPTIONS: "-XX:+UnlockExperimentalVMOptions -XX:+UseCGroupMemoryLimitForHeap"
+    - GRADLE_OPTS: '-Dorg.gradle.daemon=false -Dorg.gradle.jvmargs="-XX:+HeapDumpOnOutOfMemoryError"'
```

Comment:
Could try `JVM_OPTS: -Xmx3200m` or larger if you've increased the size?
https://circleci.com/docs/2.0/language-android/

Reply:
Shouldn't be needed with `UseCGroupMemoryLimitForHeap` flag. As per Twitter discussion, we tried different combinations and none of them ended up working.

Decided to increase the container size for Android as an alternative (and free) approach. CircleCI support turned it on for our repository after my request.